modules/json/src/smithy4s/http/json/SchematicJCodec.scala

```diff
@@ -744,28 +744,50 @@ private[smithy4s] class SchematicJCodec(maxArity: Int)
       field: Field[JCodecMake, Z, A]
   ): (Z, JsonWriter) => Unit = {
     field.fold(new Field.Folder[JCodecMake, Z, (Z, JsonWriter) => Unit] {
-      val jLabel = jsonLabel(field)
       def onRequired[AA](
           label: String,
           instance: JCodecMake[AA],
           get: Z => AA
       ): (Z, JsonWriter) => Unit = {
         val codec = instance.get
-        (z: Z, out: JsonWriter) => {
-          out.writeKey(jLabel)
-          codec.encodeValue(get(z), out)
+        val jLabel = jsonLabel(field)
+        if (jLabel.forall(JsonWriter.isNonEscapedAscii)) {
```

Comment:
interesting: performing the ascii check + ascii write is faster than just writekey

Reply:
`jLabel.forall(JsonWriter.isNonEscapedAscii)` is called only during creation-time of field handlers, which then are memorized